.github/workflows/integration_test_juju3.yaml

```diff
@@ -0,0 +1,26 @@
+name: integration-tests
+
+on:
+  pull_request:
+  workflow_call:
+
+jobs:
+  integration-test-juju3:
+    name: Integration Test on Juju3
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    secrets: inherit
+    with:
+      extra-arguments: >-
+        --openstack-rc=${GITHUB_WORKSPACE}/openrc
+        --kube-config=${GITHUB_WORKSPACE}/kube-config
+        --screenshot-dir=/tmp
+      juju-channel: 3.1/stable
```

Comment:
You'll probably  need to specify the microk8s channel too `channel: 1.28-strict/stable`